I have many useful changes in this branch now, feel free to cherry pick whatever you want

intangir opened the pull request:
better fire protection
better lava protection
better for the 1.4.6 item removal issue i mentioned in another thread on cancelled messages
logging of reinforcement breaks to preciousstones snitches
tools are damaged when breaking reinforced blocks even if the block doesnt break

i made a priority change that probably has no difference for you, it was to improve compatibility with a mod you don't use

(also some housekeeping on files that i didn't want in my branch)

remyroy committed:
Adding water protection in addition to lava for rails.

Conflicts:

	src/com/untamedears/citadel/listener/BlockListener.java

intangir committed:
stop tracking this junk

intangir committed:
catch harmless database error and ignore it

intangir committed:
added aggressive fire and lava grief protection

intangir committed:
fixed the error i made when i changed remys change

intangir committed:
Merge branch 'exultant' into intangir

intangir committed:
changed priority so that the message is recieved sooner

intangir committed:
3 changes:
A more elegant and simple fix for the item removal issue that appeared with 1.4.6
Log protected block breaks on snitch block even if they still have reinforcement left
Damage tools when they are used on protected blocks

ttk2 commented:
Cool, Cool, I will have Erocs review the pull just because he has been doing maintenance then merge it and put it up for a short testing before placing it online. Thanks for your work man.

intangir commented:
Some of these changes may not be ideal as is for civcraft 
The tool damaging one adds a lot of expense to breaking things, it works out well on my server because they can be repaired and people don't have 20 tick dro vaults ;) but civcraft is another story

removing diamond tools from the damaged list might be prudent

erocs commented:
I made some quick comments, likely more as notes to myself. I like the tool damage, but it may need some adjustment for Civcraft. Thanks!

intangir commented:
Ya there is already rioting over it, might want to exclude damage to diamond

intangir committed:
fixed a duplication, also added consideration for unbreaking tools

intangir committed:
added default protection level, reinforcing by hand, auto bypassing and uprading of default protections
also bumped up the protection stats
and added a couple more securables

intangir committed:
allow fortify mode by hand also (helps with groups), default security to private

intangir committed:
made a flag to hide info from people who dont have access on the block (defaults to hide)

intangir committed:
added reinforcing while in info mode also, made pistons bypass default protection, tuned up the extend code a bit

intangir commented:
I added a new feature that you probably wont want on civcraft, but its pretty cool nonetheless

a default level of protection, "by hand" protection, (uses material 'air' hehe)
it also auto bypasses your own 'by hand' protections
allows reinforcement without issue
you can change group on them too
it also auto removes by hand protection when they are pushed by a piston

i also added a feature to allow hiding of /ctinfo information if your using it on blocks you don't have access to

they can be left off, or turned on based on configuration options
shouldn't effect anything if you leave the features off

erocs commented:
For the "by hand" protection, it looks like a free low-grade low-annoyance reinforcement, correct? Pretty cool. We'll have to ask ttk2 but I'm suspecting he won't want to enable it.

We should cherry-pick ed25241, for the /ctinfo change.

ttk2 commented:
We already add nearly 60,000 reinforcements per day there is no reason to
increase that.
On Feb 22, 2013 12:13 PM, "erocs" notifications@github.com wrote:

> For the "by hand" protection, it looks like a free low-grade low-annoyance
> reinforcement, correct? Pretty cool. We'll have to ask ttk2 but I'm
> suspecting he won't want to enable it.
> 
> We should cherry-pick ed25241https://github.com/Exultant/Citadel/commit/ed25241,
> for the /ctinfo change.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/Exultant/Citadel/pull/99#issuecomment-13960322.

intangir commented:
might want to include it anyway and leave the option off for civcraft

if the mod is publicly released it is a good idea to offer more options

the reason i wanted it is because im having a hard time keeping new players, even harder when before they learn how to protect their things they are all burned down or griefed

the free protection prevents burning

also i made a quick youtube video explaining the basics of citadel usage since so few of my newer players ever bother to reinforce their stuff.., it didnt help though ;)

i can link it later (dont have access to youtube at work)

intangir committed:
fix for by hand protection when toggling out of group mode, it must reset state faction also

intangir committed:
cancel all transfers to hoppers, at least for now

intangir committed:
added alot of new types to config, changed durability settings back

intangir committed:
improved chest, trapped chest, and hopper support

intangir committed:
multiple versions of updates, brings it upto 1.9+
